Library/Homebrew/extend/os/mac/requirements/java_requirement.rb

```diff
@@ -13,7 +13,8 @@ def possible_javas
     javas << Pathname.new(ENV["JAVA_HOME"])/"bin/java" if ENV["JAVA_HOME"]
     javas << java_home_cmd
     javas << which("java")
-    javas
+    javas.delete(Pathname.new("/usr/bin/java")) # /usr/bin/java is a stub on macOS
+    javas.compact
```

Comment:
Instead of adding and deleting let's do:
```ruby
javas << java_home_cmd
which_java = which("java")
# /usr/bin/java is a stub on macOS
javas << which_java unless which_java.to_s == "/usr/bin/java"
javas
```